app/javascript/runtimeBanner/RuntimeBanner.jsx

```diff
@@ -0,0 +1,146 @@
+/* global Runtime */
+
+import { h, Component } from 'preact';
+
+const BANNER_DISMISS_KEY = 'runtimeBannerDismissed';
+
+/**
+ * A banner that will be displayed to provide a deep link into a specified
+ * ConsumerApp based on the Runtime context. If the banner is dismissed it will
+ * keep track of this in localStorage so it won't be rendered again.
+ */
+export class RuntimeBanner extends Component {
+  dismissBanner = () => {
+    localStorage.setItem(BANNER_DISMISS_KEY, true);
+    this.removeFromDOM();
+  };
+
+  removeFromDOM = () => {
+    const container = document.getElementById('runtime-banner-container');
+    container?.remove();
+  };
+
+  handleDeepLinkFallback = () => {
+    // These buttons should exist in the DOM in order to attempt the fallback
+    // mechanism (they only exist in the path `/r/mobile`)
+    const installNowButton = document.getElementById('link-to-mobile-install');
+    const retryButton = document.getElementById('link-to-mobile-install-retry');
+    if (!installNowButton || !retryButton) {
+      return;
+    }
+
+    // The target path comes in via GET request query params
+    const urlParams = new URLSearchParams(window.location.search);
+    const targetPath = urlParams.get('deep_link');
+
+    // Constants - they will become dynamic (configurable by creators) in upcoming releases
+    const FOREM_IOS_SCHEME = 'com.forem.app';
+    const FOREM_APP_STORE_URL =
+      'https://apps.apple.com/us/app/forem/id1536933197';
+    const FOREM_GOOGLE_PLAY_URL =
+      'https://play.google.com/store/apps/details?id=to.dev.dev_android';
+
+    if (Runtime.currentOS() === 'iOS') {
+      // The install now must target Apple's AppStore
+      installNowButton.href = FOREM_APP_STORE_URL;
+
+      // We try to deep link directly by launching a custom scheme and populate
+      // the retry button in case the user will need it
+      const targetLink = `${FOREM_IOS_SCHEME}://${window.location.host}${targetPath}`;
+      retryButton.href = targetLink;
+      window.location.href = targetLink;
+    } else if (Runtime.currentOS() === 'Android') {
+      const targetIntent =
+        'intent://scan/#Intent;scheme=zxing;package=com.google.zxing.client.android;end';
+      retryButton.href = targetIntent;
+      installNowButton.href = FOREM_GOOGLE_PLAY_URL;
+
+      // Android support will not be available yet. Links to `/r/mobile` won't be
+      // visible in Android browsers. However, as a fallback (safety) measure to
+      // avoid having Android users land in this page we redirect them back to the
+      // home page. This to avoids users landing in an unsupported (not working
```

Comment:
I think this comment got a bit garbled: `This to avoids users landing in an unsupported...`

Reply:
Thanks for spotting this! I can definitely see this as well. Too many edits turned it into a "blob" of ideas mushed together. I'll definitely re-write this one.